s3/src/main/java/com/yahoo/ycsb/db/S3Client.java

```diff
@@ -0,0 +1,348 @@
+/**
+ * S3 storage client binding for YCSB.
+ *
+ * Submitted by Ivan Baldinotti on 14/07/2015
+ *
+ */
+package com.yahoo.ycsb.db;
+
+import java.util.HashMap;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import com.yahoo.ycsb.ByteArrayByteIterator;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.*;
+import com.amazonaws.auth.*;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+/**
+ * S3 Storage client for YCSB framework.
+ * 
+ * Properties to set:
+ * 
+ * s3.accessKeyId=access key S3 aws
+ * s3.secretKey=secret key S3 aws
+ * s3.endPoint=s3.amazonaws.com
+ * s3.region=us-east-1
+ * The parameter table is the name of the Bucket where to upload the files. 
+ * This must be created before to start the benchmark
+ * The size of the file to upload is determined by two parameters:
+ * - fieldcount this is the number of fields of a record in YCSB
+ * - fieldlength this is the size in bytes of a single field in the record
+ * together these two parameters define the size of the file to upload, 
+ * the size in bytes is given by the fieldlength multiplied by the fieldcount.
+ * The name of the file is determined by the parameter key. 
+ *This key is automatically generated by YCSB.
+ *
+ * @author ivanB1975
+ */
+public class S3Client extends DB {
+ 
+  private static String accessKeyId;
```

Comment:
it looks like most of these variables aren't needed outside of the code block where we initialize the s3Client. You should limit all of them to that scope and remove them from the set of static members.